corehq/apps/translations/forms.py

```diff
@@ -271,3 +271,43 @@ class TransifexOrganizationForm(forms.ModelForm):
     def __init__(self, *args, **kwargs):
         super(TransifexOrganizationForm, self).__init__(*args, **kwargs)
         self.initial['api_token'] = b64_aes_decrypt(self.instance.api_token)
+
+
+class AddTransifexBlacklistForm(forms.ModelForm):
+    app_id = forms.ChoiceField(label=ugettext_lazy("Application"), choices=(), required=True)
+    action = forms.CharField(widget=forms.HiddenInput)
+    domain = forms.CharField(widget=forms.HiddenInput)
+
+    def __init__(self, domain, *args, **kwargs):
+        super(AddTransifexBlacklistForm, self).__init__(*args, **kwargs)
+        self.helper = hqcrispy.HQFormHelper()
+
+        self.fields['app_id'].choices = tuple((app.id, app.name) for app in get_brief_apps_in_domain(domain))
+        form_fields = [
+            hqcrispy.Field('app_id'),
+            hqcrispy.Field('module_id'),
```

Comment:
you dont need to add `module_id` and couple of others on the form as properties above?

Should this form validate things like
1. module_id passed is actually a module in the app?
2. maybe, display_text/field name is actually present in the module?

Reply:
https://github.com/dimagi/commcare-hq/pull/23429/commits/60a03f62238db5a87a26358ab8c69bd66a2fcd17

> maybe, display_text/field name is actually present in the module?
 I think that's a bit harder as ICDS/REACH uses some custom case list XML. We don't have the ability to parse that at the moment (which is why its custom) so we wouldn't be able to confidently validate that

Reply:
> you dont need to add module_id and couple of others on the form as properties above?

under the `Meta` class, it has `fields = __all__` You only need to add at the class level if you want to override them